library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,573 @@
+  # socket (or path or timer). This class is able to group those units and offer an API
+  # to handle them together.
+  #
+  # See also {file:library/systemd/doc/services_and_sockets.md}.
+  #
+  # @note All changes performed over an object of this class are not applied into the
+  # underlying system until the {#save} method is called.
+  #
+  # @example Enabling a service
+  #   cups = SystemService.find("cups")
+  #   cups.start_mode = :on_boot
+  #   cups.save
+  #
+  # @example Activating a service
+  #   cups = SystemService.find("cups")
+  #
+  #   cups.currently_active? #=> false
+  #   cups.active?           #=> false
+  #
+  #   cups.start
+  #
+  #   cups.currently_active? #=> false
+  #   cups.active?           #=> true
+  #
+  #   cups.save
+  #
+  #   cups.currently_active? #=> true
+  #   cups.active?           #=> true
+  #
+  # @example Changing start mode
+  #   cups = SystemService.find("cups")
+  #   cups.start_mode = :on_demand
+  #
+  #   cups.current_start_mode #=> :on_boot
+  #   cups.start_mode         #=> :on_demand
+  #
+  #   cups.save
+  #
+  #   cups.current_start_mode #=> :on_demand
+  #   cups.start_mode         #=> :on_demand
+  #
+  # @example Ignoring status changes (useful when changing the service on 1st stage)
+  #   cups = SystemService.find("cups")
+  #   cups.start
+  #
+  #   cups.currently_active? #=> false
+  #   cups.active?           #=> true
+  #
+  #   cups.save(keep_state: true)
+  #
+  #   cups.currently_active? #=> false
+  class SystemService
+    extend Forwardable
+
+    # Error when a service is not found
+    class NotFoundError < RuntimeError; end
+
+    # @return [Yast::SystemdServiceClass::Service]
+    attr_reader :service
+
+    # @return [Hash<Symbol, Object>] Errors when trying to write changes to the underlying system.
+    #   * :active [Boolean] whether the service should be active after saving
+    #   * :start_mode [Symbol] start mode the service should have after saving
+    attr_reader :errors
+
+    # @return [Symbol, nil] :start, :stop, :restart or :reload. It returns nil
+    #   if no action has been requested yet.
+    attr_reader :action
+
+    # @!method support_reload?
+    #
+    # @return [Boolean]
+    def_delegator :@service, :can_reload?, :support_reload?
+
+    # @!method name
+    #   @see Yast::SystemdServiceClass::Service#name
+    #   @return [String]
+    # @!method static?
+    #   @see Yast::SystemdServiceClass::Service#static?
+    #   @return [Boolean]
+    # @!method running?
+    #   @see Yast::SystemdServiceClass::Service#running?
+    #   @return [Boolean]
+    # @!method description
+    #   @see Yast::SystemdServiceClass::Service#description
+    #   @return [String]
+    def_delegators :@service, :name, :static?, :running?, :description
+
+    class << self
+      # Finds a service by its name
+      #
+      # @param name [String] service name with or without extension (e.g., "cups" or "cups.service")
```

Comment:
So... if "cups.service" works, does "cups.socket" work too? That is, am I able to find the SystemService using any of its unit names? If not, wouldn't it be better to allow only the short name?

Reply:
I don't think so. Both, "cups" and "cups.service" work because SystemdServiceClass#find supports it. `ServiceManager` simply delegates the call, see https://github.com/yast/yast-yast2/blob/master/library/systemd/src/modules/systemd_service.rb#L77